src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -417,38 +418,56 @@ void LottieBuilder::updateRect(LottieGroup* parent, LottieObject** child, float
     auto size = rect->size(frameNo, tween, exps);
     auto pos = rect->position(frameNo, tween, exps) - size * 0.5f;
     auto r = rect->radius(frameNo, tween, exps);
+    auto modifier = ctx->modifier;
 
     if (r == 0.0f)  {
-        if (ctx->roundness) ctx->roundness->modifyRect(size, r);
+        //roundness is the first modifier -> it can be applied before the shape's path is established
```

Comment:
I think this can be delegated to the relying modifier itself, allowing it to properly handle the logic internally by receiving the current shape type.

ie:
`modifyPath(... modifying shape type : rect/polygon/path/ellipse/start/etc)`

you can also improve all the if-else condition other places with the above approach.